src/Polly.Core/Retry/RetryResilienceStrategy.cs

```diff
@@ -53,6 +53,11 @@ protected internal override async ValueTask<Outcome<T>> ExecuteCore<TState>(Func
         {
             var startTimestamp = _timeProvider.GetTimestamp();
             var outcome = await StrategyHelper.ExecuteCallbackSafeAsync(callback, context, state).ConfigureAwait(context.ContinueOnCapturedContext);
+            if (context.CancellationToken.IsCancellationRequested)
+            {
+                outcome = Outcome.FromException<T>(new OperationCanceledException(context.CancellationToken));
```

Comment:
I think it would be better to acknowledge cancellation _only when_ the retry policy would handle the outcome. It's important that a _valid_ outcome be surfaced by the policy, since the choice to continue processing may well have been deliberate (cf. item three of [MS recommendations](https://devblogs.microsoft.com/premier-developer/recommended-patterns-for-cancellationtoken/)).

Reply:
Before this PR:
- If the cancellation is requested before the execution of a given attempt (either the original or any handled retry attempt) then it will short cut the execution with an OCE.
- If the cancellation is requested during the `OnRetry` then it will short cut the execution with an OCE.
- If the cancellation is requested during the waiting of the retry delay then it will short cut the execution with an OCE.

This PR handles the following case:
- If the cancellation is requested during the execution of the user callback then regardless of the outcome then it will short cut the execution with an OCE.

Just to clarify: are you asking to
- return the outcome of the user callback regardless of the cancellation was requested (or not) if the strategy won't handle the outcome of the user callback
- return the outcome of the user callback if the strategy would handle the outcome and it was the last attempt
- short cut the execution with an OCE if the strategy would handle the outcome it was not the last attempt

Is my understanding correct?